hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -265,7 +312,33 @@ static void configRtcWakeupSource(const hal_wakeup_source_base_t* wakeupSources)
     }
 }
 
-static bool isWokenUpByGpio(hal_wakeup_source_gpio_t* gpioWakeup) {
+static void configUsartWakeupSource(const hal_wakeup_source_base_t* wakeupSources) {
+    auto source = wakeupSources;
+    while (source) {
+        if (source->type == HAL_WAKEUP_SOURCE_TYPE_USART) {
+            nrf_uarte_int_disable(NRF_UARTE0, NRF_UARTE_INT_RXDRDY_MASK);
+            nrf_uarte_event_clear(NRF_UARTE0, NRF_UARTE_EVENT_RXDRDY);
+            nrf_uarte_int_enable(NRF_UARTE0, NRF_UARTE_INT_RXDRDY_MASK);
+            NVIC_EnableIRQ(UARTE0_UART0_IRQn);
+        }
+        source = source->next;
+    }
+}
+
+static void configNetworkWakeupSource(const hal_wakeup_source_base_t* wakeupSources) {
+    auto source = wakeupSources;
+    while (source) {
+        if (source->type == HAL_WAKEUP_SOURCE_TYPE_NETWORK) {
+            nrf_uarte_int_disable(NRF_UARTE1, NRF_UARTE_INT_RXDRDY_MASK);
+            nrf_uarte_event_clear(NRF_UARTE1, NRF_UARTE_EVENT_RXDRDY);
```

Comment:
This may cause problems for code that is blocked on USART waiting for incoming data. We've cleared an event despite potentially having received data already leaving the thread waiting on the data still in a blocked state.